### ISSUE TITLE
Cannot install without the Git repository

### ISSUE THREAD

**Issue by paugier** [hpyproject/hpy#92]
Sorry to bother with such a silly thing.

In `setup.py` (https://github.com/hpyproject/hpy/blob/master/setup.py#L18), there is 
```
    gitrev = subprocess.check_output('git describe --abbrev=7 --dirty '
                                     '--always --tags --long'.split(), encoding='utf-8')
```
so that hpy cannot be installed from source without the Git repository.

It seems quite uncommon because I'm able to use hg-git for many Python related projects on Github.

I modified `setup.py` to support also hg-git (https://github.com/hpyproject/hpy/compare/master...paugier:do-not-assume-git-repo). Do you think something like that could be included in master (yes it is a bit specific :slightly_smiling_face:)? Or could there be another solution to install hpy from source without the Git repository?

Is the Git revision use for anything important in HPy?

**Comment by arigo** [comment]
+1 to include this in master.

**Comment by hodgestar** [comment]
@paugier Thank you for pointing this out -- it is indeed a bit of a mess that needs tidying!

**Comment by hodgestar** [comment]
After some discussion in IRC, the consensus seems to be to apply a simpler fix for now. The simpler fix ignores failures and writes a fixed "__UNKNOWN__" git revision if one is not found. I've implemented this in #94, so I'll close this PR.

**Comment by hodgestar** [comment]
@paugier I've merged #94. Please re-open this issue if this still fails for you.
